### PR TITLE
Feature/author profile

### PR DESCRIPTION
**Pull request by thnukid**
# Author profiles with post links
* Page with all posts by an author =>` /author/name-of-author.html`
* Page of all authors for the guild => `/authors`
* Navbar link to `/authors`
* Links form a blog post to the author page

Partially addresses issue #45

### CODE REVIEW

File: helpers/author_helpers.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      value.gsub!(/[']+/, '')
+      value.gsub!(/\W+/, ' ')
+      value.strip!
+      value.downcase!
+      value.gsub!(' ', '-')
+      value
+    end
+  end
+
+  def pluralize(singular, count, plural = nil)
```
Comment (bazzel) [inline helpers/author_helpers.rb]: `pluralize` is available in [Padrino](https://devdocs.io/padrino/padrino/helpers/formathelpers#pluralize-instance_method). Don't think we need it here?

File: helpers/author_helpers.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  def authors
+    data.site.author.values.sort_by(&:name)
+  end
+
+  # Returns a slug for any string
+  #
+  # @example
+  #   "Some String To Convert".to_slug => "some-string-to-convert"
+  String.class_eval do
+    def to_slug
```
Comment (bazzel) [inline helpers/author_helpers.rb]: I prefer using the same approach as we did for kabisa.nl and add a custom mapper for the author. (https://github.com/kabisa/kabisa.nl/blob/master/lib/mappers/author_mapper.rb)

File: source/authors/index.html.slim [hunk 3]
```diff
@@ -0,0 +1,19 @@
+---
+layout: landing
+---
+
+section
+  .flex-posts-list
+    - authors.each do |author|
```
Comment (bazzel) [inline source/authors/index.html.slim]: Looks fine, but personally I prefer extracting the template code for a single item into a partial, but that's a personal preference.

**Comment by bazzel** [comment]
Position of image is not correct.

Screenshot_2019-12-11_at_10_40_51

**Comment by pascalw** [comment]
CI is failing due to mixed content. Not sure where this vimeo player is coming from?

File: source/authors/index.html.slim (hunk 3 above)
Comment (thnukid) [reply source/authors/index.html.slim]: added
```
= partial 'partials/author-teaser', locals: { author: author, post_count:  post_count }
```

File: helpers/author_helpers.rb (hunk 1 above)
Comment (thnukid) [reply helpers/author_helpers.rb]: Good one! Using Padrino helper

File: helpers/author_helpers.rb (hunk 2 above)
Comment (thnukid) [reply helpers/author_helpers.rb]: Added custom mapper for Author and Post (because Post also contains author array)

**Comment by thnukid** [comment]
Screenshot 2019-12-11 at 12 29 59
👨‍🚀

**Comment by pascalw** [comment]
@thnukid I noticed only the upper left part of the author image is clickable, can you look into this?

hover

**Comment by pascalw** [comment]
With regards to the avatars, it does seem that the avatars don't stretch to fill the full avatar container:

Screenshot 2019-12-11 at 15 06 57

In this case the container is ~200x200px, but the image is ~400x400 so it should be able to fill it fully.

**Comment by thnukid** [comment]
Todo: Avatars that are not 200x200 or similar

[ ] https://5df1051aedd1440008386027--theguild-preview.netlify.com/authors/andy-maes/
[ ] https://5df1051aedd1440008386027--theguild-preview.netlify.com/authors/etienne-van-delden/
[ ] https://5df1051aedd1440008386027--theguild-preview.netlify.com/authors/jonny-kievits/
[ ] https://5df1051aedd1440008386027--theguild-preview.netlify.com/authors/luc-tielen/
[ ] https://5df1051aedd1440008386027--theguild-preview.netlify.com/authors/nicole-bekkers/
[ ] https://5df1051aedd1440008386027--theguild-preview.netlify.com/authors/niek-horsmans/
[ ] https://5df1051aedd1440008386027--theguild-preview.netlify.com/authors/tjaco-oostdijk/
[ ] https://5df1051aedd1440008386027--theguild-preview.netlify.com/authors/tonci-galic/

@pascalw 
* Good catch with the `tiles` of authors # fixed
* Made the author image full size, and the above todo to replace the images with something square

**Comment by pascalw** [comment]
@thnukid Can I suggest this change?

```patch
diff --git a/source/stylesheets/site.css.scss b/source/stylesheets/site.css.scss
index 5d3a1ec..a5d7e68 100644
--- a/source/stylesheets/site.css.scss
+++ b/source/stylesheets/site.css.scss
@@ -159,11 +159,14 @@ main {
 
     img.author-image {
       border-radius: 50%;
+      object-fit: cover;
+      width: 100%;
     }
   }
 
   .hero-author {
     padding: 0em;
+    display: flex;
   }
 }
 .hero-big {
```

Before:

Screenshot 2019-12-11 at 16 49 13
Screenshot 2019-12-11 at 16 49 22
Screenshot 2019-12-11 at 16 49 29

After:
Screenshot 2019-12-11 at 16 49 10
Screenshot 2019-12-11 at 16 49 16
Screenshot 2019-12-11 at 16 49 25

**Comment by thnukid** [comment]
@pascalw I have added your suggest about object-fit, also added some space between author image and text.

@bazzel Do I get approved? 🚀

**Comment by bazzel** [comment]
> @pascalw I have added your suggest about object-fit, also added some space between author image and text.
> 
> @bazzel Do I get approved? 🚀

👍

**Pull request merged**